File: LC-Python-Solution/682. Baseball Game.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:

        res = []

        for item in operations:
            if item == '+':
                temp = res[-1]+res[-2]
                res.append(temp)
            
            elif item == 'D':
                temp = res[-1]*2
                res.append(temp)
            
            elif item == 'C':
                res.pop()
            
            else:
                res.append(int(item))
                
        return sum(res)
            




        

File: LC_JS-SOLUTIONS/682. Baseball Game.js
/**
 * @param {string[]} operations
 * @return {number}
 */
var calPoints = function (operations) {
  let res = [];

  for (let i = 0; i < operations.length; i++) {
    if (operations[i] === "+") {
      res.push(res[res.length - 1] + res[res.length - 2]);
    } else if (operations[i] === "D") {
      res.push(res[res.length - 1] * 2);
    } else if (operations[i] === "C") {
      res.pop();
    } else {
      res.push(Number(operations[i]));
    }
  }

  return res.reduce((a, b) => a + b, 0);
};
